core/sqf/src/seatrans/tm/hbasetmlib2/src/main/java/org/trafodion/dtm/HBaseTxClient.java

```diff
@@ -1034,34 +1034,23 @@ public void run() {
-                                      zookeeper.deleteRegionEntry(regionEntry);
-                                      // Create a local HTable object using the regionInfo
-                                      HTable table = new HTable(config, HRegionInfo.parseFrom(regionBytes).getTable().getNameAsString());
-                                      // Repost a zookeeper entry for all current regions in the table
-                                      zookeeper.postAllRegionEntries(table);
-                                }// NotServingRegionException
-                                catch (TableNotFoundException tnfe) {
-                                   // In this case there is nothing to recover.  We just need to delete the region entry.
-                                      // First delete the zookeeper entry
-                                      LOG.warn("TRAF RCOV THREAD:TableNotFoundException calling txnManager.recoveryRequest. " + "TM: " +
-                                              tmID + " regionBytes: [" + regionBytes + "].  Deleting zookeeper region entry. \n exception: " + tnfe);
-                                      zookeeper.deleteRegionEntry(regionEntry);
-
-                                }// TableNotFoundException
-                                catch (DeserializationException de) {
-                                   // We are unable to parse the region info from ZooKeeper  We just need to delete the region entry.
-                                      // First delete the zookeeper entry
-                                      LOG.warn("TRAF RCOV THREAD:DeserializationException calling txnManager.recoveryRequest. " + "TM: " +
-                                              tmID + " regionBytes: [" + regionBytes + "].  Deleting zookeeper region entry. \n exception: " + de);
-                                      zookeeper.deleteRegionEntry(regionEntry);
-                                }// DeserializationException
+                                }
+                                catch (IOException e) {
+                                   // For all cases of Exception, we rely on the region to redrive the request.
+                                   // Likely there is nothing to recover, due to a stale region entry, but it is always safe to redrive.
+                                   // We log a warning event and delete the ZKNode entry.
+                                   LOG.warn("TRAF RCOV THREAD:Exception calling txnManager.recoveryRequest. " + "TM: " +
+                                              tmID + " regionBytes: [" + regionBytes + "].  Deleting zookeeper region entry. \n exception: ", e);
+                                   zookeeper.deleteRegionEntry(regionEntry);
+
+                                   // In the case of NotServingRegionException we will repost the ZKNode after refreshing the table.
+                                   if (e instanceof NotServingRegionException){
```

Comment:
Do you think we should consider checking e.getCause() instanceof NotServingRegionException too. Rationale: Assuming recoveryRequest throws NotServingRegionException, it would be caught in 3018 and transformed into IOException with cause as NotServingRegionException.